history.go

```diff
@@ -0,0 +1,75 @@
+package merkletree
+
+import (
+	"errors"
+
+	"github.com/coniks-sys/libmerkleprefixtree-go/crypto"
+)
+
+var (
+	ErrHistoryExisted = errors.New("[merkletree] trying to override existing history")
+	ErrBadEpoch       = errors.New("[merkletree] next epoch's STR has bad epoch ")
+)
+
+type History struct {
+	tree          *MerkleTree
+	currentSTR    *SignedTreeRoot
+	epochInterval int64
+}
+
+func NewHistory(m *MerkleTree, key crypto.SigningKey, startEp, epInterval int64) *History {
+	h := new(History)
+	h.tree = m
+	h.epochInterval = epInterval
+	h.currentSTR = newSTR(m, startEp, 0, make([]byte, crypto.HashSizeByte), key)
+	return h
+}
+
+func (h *History) UpdateHistory(m *MerkleTree, nextEp int64) error {
```

Comment:
Should `nextEp` be a parameter, which is exactly the time when `UpdateHistory` is called? Or it could be computed like this `nextEp := h.nextSTR.epoch + h.nextSTR.P.epochInterval` ? 
I think the former is more proper way?